[CI] PeerRecoveryTargetServiceTests testPrepareIndexForPeerRecovery failure

```
java.lang.AssertionError: 0 > -1
	at __randomizedtesting.SeedInfo.seed([9ADBDC3EA699F935:41553D6437F6AF95]:0)
	at org.elasticsearch.index.translog.Translog.newSnapshot(Translog.java:615)
	at org.elasticsearch.indices.recovery.PeerRecoveryTargetServiceTests.testPrepareIndexForPeerRecovery(PeerRecoveryTargetServiceTests.java:180)
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+release-tests/888/console
https://gradle-enterprise.elastic.co/s/hjg6zbffzaoxu

```
./gradlew ':server:test' --tests "org.elasticsearch.indices.recovery.PeerRecoveryTargetServiceTests.testPrepareIndexForPeerRecovery" \
  -Dtests.seed=9ADBDC3EA699F935 \
  -Dtests.security.manager=true \
  -Dbuild.snapshot=false \
  -Dtests.jvm.argline="-Dbuild.snapshot=false" \
  -Dtests.locale=es-HN \
  -Dtests.timezone=Africa/Dakar \
  -Dcompiler.java=13
```

--- a/server/src/test/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetServiceTests.java
+++ b/server/src/test/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetServiceTests.java
@@ -177,11 +177,13 @@ public class PeerRecoveryTargetServiceTests extends IndexShardTestCase {
         Optional<SequenceNumbers.CommitInfo> safeCommit = shard.store().findSafeIndexCommit(globalCheckpoint);
         assertTrue(safeCommit.isPresent());
         int expectedTotalLocal = 0;
-        try (Translog.Snapshot snapshot = getTranslog(shard).newSnapshot(safeCommit.get().localCheckpoint + 1, globalCheckpoint)) {
-            Translog.Operation op;
-            while ((op = snapshot.next()) != null) {
-                if (op.seqNo() <= globalCheckpoint) {
-                    expectedTotalLocal++;
+        if (safeCommit.get().localCheckpoint < globalCheckpoint) {
+            try (Translog.Snapshot snapshot = getTranslog(shard).newSnapshot(safeCommit.get().localCheckpoint + 1, globalCheckpoint)) {
+                Translog.Operation op;
+                while ((op = snapshot.next()) != null) {
+                    if (op.seqNo() <= globalCheckpoint) {
+                        expectedTotalLocal++;
+                    }
                 }
             }
         }

